framework/include/auxkernels/MaterialStdVectorAuxBase.h

```diff
@@ -0,0 +1,60 @@
+/****************************************************************/
+/*               DO NOT MODIFY THIS HEADER                      */
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*           (c) 2010 Battelle Energy Alliance, LLC             */
+/*                   ALL RIGHTS RESERVED                        */
+/*                                                              */
+/*          Prepared by Battelle Energy Alliance, LLC           */
+/*            Under Contract No. DE-AC07-05ID14517              */
+/*            With the U. S. Department of Energy               */
+/*                                                              */
+/*            See COPYRIGHT for full restrictions               */
+/****************************************************************/
+
+#ifndef MATERIALSTDVECTORAUXBASE_H
+#define MATERIALSTDVECTORAUXBASE_H
+
+// MOOSE includes
+#include "MaterialAuxBase.h"
+
+// Forward declarations
+template<typename T = Real>
+class MaterialStdVectorAuxBase;
+
+template<>
+InputParameters validParams<MaterialStdVectorAuxBase<> >();
+/**
+ * A base class for the various Material related AuxKernal objects
+ */
+template<typename T>
+class MaterialStdVectorAuxBase : public MaterialAuxBase<std::vector<T> >
+{
+public:
+
+  /**
+   * Class constructor
+   * @param name Name of the AuxKernel
+   * @param parameters The input parameters for this object
+   */
+  MaterialStdVectorAuxBase(const std::string & name, InputParameters parameters);
+
+  /**
+   * Class destructor
+   */
+  virtual ~MaterialStdVectorAuxBase(){}
+
+protected:
+  virtual Real computePropertyValue() = 0;
+  /// index of the vecor element
+  unsigned int _index;
+};
+
+template<typename T>
+MaterialStdVectorAuxBase<T>::MaterialStdVectorAuxBase(const std::string & name, InputParameters parameters) :
+    MaterialAuxBase<std::vector<T> >(name, parameters),
+    _index(this->template getParam<unsigned int>("index"))
```

Comment:
No! We explicitly call the base class version!

On Thu, Jul 2, 2015, 1:18 PM SudiptaBiswas <notifications@github.com> wrote:

> In framework/include/auxkernels/MaterialStdVectorAuxBase.h
> <https://github.com/idaholab/moose/pull/5331#discussion_r33812853>:
>
> > +
> > +  /**
> > +   * Class destructor
> > +   */
> > +  virtual ~MaterialStdVectorAuxBase(){}
> > +
> > +protected:
> > +  virtual Real computePropertyValue() = 0;
> > +  /// index of the vecor element
> > +  unsigned int _index;
> > +};
> > +
> > +template<typename T>
> > +MaterialStdVectorAuxBase<T>::MaterialStdVectorAuxBase(const std::string & name, InputParameters parameters) :
> > +    MaterialAuxBase<std::vector<T> >(name, parameters),
> > +    _index(this->template getParam<unsigned int>("index"))
>
> Yeah since we have a computePropertyValue now, we could implement
> computeValue in this class. This would probably overwrite the computeValue
> of MaterialAuxBase, we will have to take into account factor and offset
> here again.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/idaholab/moose/pull/5331/files#r33812853>.
>